src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -0,0 +1,1791 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.Semantics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using System;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    partial class BoundExpression : IExpression
+    {
+        ITypeSymbol IExpression.ResultType
+        {
+            get { return this.Type; }
+        }
+
+        OperationKind IOperation.Kind
+        {
+            get { return this.ExpressionKind; }
+        }
+
+        object IExpression.ConstantValue
```

Comment:
The roots of this change set are so old that this code predates the availability of those language mechanisms. I'll modernize the look of the code a bit...